docs/receptor_mesh.md

```diff
@@ -0,0 +1,68 @@
+## Receptor Mesh
+
+AWX uses a Receptor mesh to transmit "user-space" unified jobs:
+ - jobs
+ - ad hoc commands
+ - inventory updates
+
+to the node where they run.
+
+https://github.com/ansible/receptor
+
+> NOTE: user-space jobs are what carry out the user's Ansible automation. These job types run inside of the designated execution environment so that the needed content is available.
+
+### Split of Control Plane versus Execution Plane
+
+Instances in the control plane run persistent AWX services (like the web server, task dispatcher, etc.), project updates, and management jobs.
+
+Hybrid instances in the control plane may also run user-space jobs.
+
+The unified jobs API reports `controller_node` and `execution_node` fields.
+The execution node is where the job runs. The controller node prepares the private_data_dir for the job, and processes the receptor output stream.
+
+#### Receptor Configuration
+
+Execution nodes (all nodes that can run jobs) need to advertise the "ansible-runner" work type.
+All user-space jobs are submitted as a receptor work unit with this work type.
+
+An entry like this should appear in its `receptor.conf` (receptor configuration file):
+
+```
+- work-command:
+    worktype: ansible-runner
+    command: ansible-runner
+    params: worker
+    allowruntimeparams: true
+```
+
+Control (and hybrid) nodes need to advertise the "local" work type.
+Project updates are submitted as this work type.
+
+```
+- work-command:
+    worktype: local
+    command: ansible-runner
+    params: worker
+    allowruntimeparams: true
+```
+
+If user-space jobs run on a hybrid node, they will also run as the "local" work type.
```

Comment:
Wait, then how is a hybrid node different from a control-only node?  Purely the database record?  That seems off to me.

Reply:
control-only nodes will not be sent user-space jobs by the task manager. The receptor configuration, however, will be indistinguishable between the two types in this proposal.

I will try to find some place to use more specific wording to distinguish the two types.

Reply:
Ok, good to know.